stack-deploy/src/instance_spec.rs

```diff
@@ -242,12 +288,47 @@ impl InstanceSpec {
     }
 
     fn capabilities(&self) -> Vec<aws_sdk_cloudformation::types::Capability> {
-        self.capabilities.iter().cloned().collect()
+        self.instance_spec.capabilities.iter().cloned().collect()
     }
 
-    fn template_body(&self) -> &String {
-        match &self.template {
-            Template::Plain { body, .. } => body,
+    fn template_rendered(&self) -> &TemplateRendered {
+        match &self.instance_spec.template {
+            Template::Plain { rendered, .. } => rendered,
+        }
+    }
+
+    async fn set_create_template(
+        &self,
+        request: aws_sdk_cloudformation::operation::create_stack::builders::CreateStackFluentBuilder,
+    ) -> aws_sdk_cloudformation::operation::create_stack::builders::CreateStackFluentBuilder {
+        match self.upload_template().await {
+            Ok(template_url) => request.template_url(template_url),
+            Err(template_body) => request.template_body(template_body),
+        }
+    }
+
+    async fn set_update_template(
+        &self,
+        request: aws_sdk_cloudformation::operation::update_stack::builders::UpdateStackFluentBuilder,
+    ) -> aws_sdk_cloudformation::operation::update_stack::builders::UpdateStackFluentBuilder {
+        match self.upload_template().await {
+            Ok(template_url) => request.template_url(template_url),
+            Err(template_body) => request.template_body(template_body),
+        }
+    }
```

Comment:
__comment:__ At first I was going to ask why the bodies were the same, but I see we're using different `request` input args to compute the return values.